photon-client/src/integTest/scala/com/linkedin/photon/ml/estimators/GameEstimatorTest.scala

```diff
@@ -46,83 +44,177 @@ class GameEstimatorTest extends SparkTestUtils with GameTestUtils {
+   * A trivial set of fixed labeled points for simple tests to verify by hand.
+   *
+   * @return A single set of 10 vectors of 2 features and a label.
+   */
+  @DataProvider
+  def trivialLabeledPoints(): Array[Array[Seq[LabeledPoint]]] =
+
+    Array(Array(Seq(
+      LabeledPoint(0.0, Vectors.dense(-0.7306653538519616, 0.0)),
+      LabeledPoint(1.0, Vectors.dense(0.6750417712898752, -0.4232874171873786)),
+      LabeledPoint(1.0, Vectors.dense(0.1863463229359709, -0.8163423997075965)),
+      LabeledPoint(0.0, Vectors.dense(-0.6719842051493347, 0.0)),
+      LabeledPoint(1.0, Vectors.dense(0.9699938346531928, 0.0)),
+      LabeledPoint(1.0, Vectors.dense(0.22759406190283604, 0.0)),
+      LabeledPoint(1.0, Vectors.dense(0.9688721028330911, 0.0)),
+      LabeledPoint(0.0, Vectors.dense(0.5993795346650845, 0.0)),
+      LabeledPoint(0.0, Vectors.dense(0.9219423508390701, -0.8972778242305388)),
+      LabeledPoint(0.0, Vectors.dense(0.7006904841584055, -0.5607635619919824)))))
+
+  /**
+   * A very simple test that fits a toy data set using only the GAMEEstimator (not the full Driver).
+   * This is useful to understand the minimum setting in which a GAMEEstimator will function properly,
+   * and to verify the algorithms manually.
    *
-   * NOTE 1:
-   *   Intercepts are optional in GameEstimator, but GameDriver will setup an intercept by default if
-   *   none is specified in GameParams.featureShardIdToInterceptMap.
-   *   This happens in GameDriver.prepareFeatureMapsDefault, and there only.
-   *   Here, we have to setup an intercept manually, otherwise GameEstimator learns only a dependence on the
+   * @note
+   *   Intercepts are optional in GAMEEstimator, but GAMEDriver will setup an intercept by default if
+   *   none is specified in GAMEParams.featureShardIdToInterceptMap.
+   *   This happens in GAMEDriver.prepareFeatureMapsDefault, and there only.
+   *   Here, we have to setup an intercept manually, otherwise GAMEEstimator learns only a dependence on the
    *   features.
    */
-  @Test
-  def simpleFixedEffectTest(): Unit = sparkTest("simpleFixedEffectTest") {
+  @Test(dataProvider = "trivialLabeledPoints")
+  def simpleHardcodedTest(labeledPoints: Seq[LabeledPoint]): Unit = sparkTest("simpleHardcodedTest") {
 
     // This example has only a single fixed effect
     val (coordinateId, featureShardId) = ("global", "features")
-    // We will use 10 points in dimension 2 (the third feature index for the intercept)
-    val (nSamples, nDimensions) = (10, 3) // including intercept in nDimensions
+    val nDimensions = labeledPoints.head.features.size
 
     // Setup feature names a feature index from feature name to feature index
-    val featureNames = (0 until nDimensions-1).map(i => s"feature-$i").toSet
-    val featureIndexMap = DefaultIndexMap(featureNames) + ((GLMSuite.INTERCEPT_NAME_TERM, nDimensions-1))
+    // Have to drop in an intercept "feature" but intercepts are turned ON by default
```

Comment:
The comment block on line 85 - 86 is confusing me, I don't understand what it's trying express